### PR TITLE
Fix some warnings

### PR DESCRIPTION
**Pull request by mathbunnyru**

### CODE REVIEW

File: libmamba/include/mamba/core/output.hpp [hunk 1]
```diff
@@ -170,7 +170,7 @@ namespace mamba
     {
     public:
 
-        MessageLogger(const char* file, int line, log_level level);
+        MessageLogger(const char* file, log_level level);
```
Comment (Hind-M) [inline libmamba/include/mamba/core/output.hpp:173]: Can you explain why remove the line here?

File: libmamba/include/mamba/solver/libsolv/unsolvable.hpp [hunk 2]
```diff
@@ -21,7 +21,7 @@ namespace solv
 
 namespace mamba
 {
-    class Palette;
+    struct Palette;
```
Comment (Hind-M) [inline libmamba/include/mamba/solver/libsolv/unsolvable.hpp:24]: What was the warning regarding this?

File: libmamba/tests/src/download/test_mirror.cpp [hunk 3]
```diff
@@ -58,7 +58,8 @@ namespace mamba::download
         TEST_CASE("PassThroughMirror")
         {
             std::unique_ptr<Mirror> mir = make_mirror("");
-            CHECK_EQ(typeid(*mir), typeid(PassThroughMirror));
+            auto& ref = *mir;
```
Comment (Hind-M) [inline libmamba/tests/src/download/test_mirror.cpp:61]: How about the warning for this change? I believe this is more related to performance but we only dereference once.

File: libmamba/src/api/list.cpp [hunk 4]
```diff
@@ -71,7 +71,7 @@ namespace mamba
             }
 
             std::regex spec_pat(regex);
-            auto all_records = std::move(prefix_data.all_pkg_mgr_records());
+            auto all_records = prefix_data.all_pkg_mgr_records();
```
Comment (Hind-M) [inline libmamba/src/api/list.cpp:74]: And this one?

File: libmamba/include/mamba/core/output.hpp (hunk 1 above)
Comment (mathbunnyru) [reply libmamba/include/mamba/core/output.hpp:173]: `m_line` is not used anywhere, we're not logging the line number

File: libmamba/include/mamba/solver/libsolv/unsolvable.hpp (hunk 2 above)
Comment (mathbunnyru) [reply libmamba/include/mamba/solver/libsolv/unsolvable.hpp:24]: `Palette` is defined as `struct`: https://github.com/mamba-org/mamba/blob/af276aeedbb0f63a40ff500832425335d88029b9/libmamba/include/mamba/core/palette.hpp#L14

There is a warning when declaration and definition are different - as far as I remember in some ABIs (MSVC, if I remember correctly) this means different mangling and could cause linking errors.

File: libmamba/tests/src/download/test_mirror.cpp (hunk 3 above)
Comment (mathbunnyru) [reply libmamba/tests/src/download/test_mirror.cpp:61]: This is not about performance, compiler didn't like that there possibly was a call with side effects in `typeid` call (dereferencing might fail).
The easiest way to fix this is to dereference in a separate line.

Comment (Hind-M) [reply libmamba/tests/src/download/test_mirror.cpp:61]: Well in that case, it still doesn't solve the problem of dereferencing possible failure.
I suggest adding instead a `static_assert` before the `CHECK_EQ`:
`static_assert(mir != nullptr);`

File: libmamba/include/mamba/core/output.hpp (hunk 1 above)
Comment (mathbunnyru) [reply libmamba/include/mamba/core/output.hpp:173]: It doesn't give a warning (now), but it's a good point, I will remove `m_file` as well - it doesn't change any behaviour.

File: libmamba/tests/src/download/test_mirror.cpp (hunk 3 above)
Comment (mathbunnyru) [reply libmamba/tests/src/download/test_mirror.cpp:61]: I can't do it, because `mir` is not known in a compile-time, so `static_assert` won't work here

Comment (Hind-M) [reply libmamba/tests/src/download/test_mirror.cpp:61]: And maybe also rename `ref` to something more explicit, `mir_ref` for example.

Comment (mathbunnyru) [reply libmamba/tests/src/download/test_mirror.cpp]: Done

**Review by Hind-M — APPROVED** [review]
Thanks!

**Pull request merged**